### PR TITLE
Fix UG and DG

### PR DESCRIPTION
**Pull request by Vanessamae23**
## Description

Fix UG and DG

## Context

Fix UG and DG

## Checklist

- [ ] I have self-reviewed my changes.
- [ ] I have added JavaDocs to all relevant methods.
- [ ] I have updated the documentation: 
  - User Guide Table of Contents, Description and Summary.
  - Developer Guide, if applicable.
- [ ] I have ran `./gradlew test` or `./gradlew check` and all checks pass.
- [ ] I have updated my project portfolio with what I have contributed.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -242,7 +242,8 @@ Adds a student to the student book.
 <br>
 - `TELEGRAM_HANDLE` : Must be a valid telegram handle ID. 
 <br>
-- `GROUP` : A string representing the group of students as shown in the list. Must not contain space.
+- `GROUP` : A string representing the group of students as shown in the list. Must only be alphanumeric and cannot 
+contain space.
```
Comment (et-irl) [inline docs/UserGuide.md:246]: spaces

File: docs/UserGuide.md [hunk 2]
```diff
@@ -369,7 +370,8 @@ Edits an existing student.
...
```
Comment (et-irl) [inline docs/UserGuide.md:374]: spaces

**Review by Choonyan02 — APPROVED** [review]
LGTM, I like the consistency!

File: docs/UserGuide.md [hunk 3]
```diff
@@ -498,7 +500,8 @@ Marks the attendance of a group of students for that tutorial.
 
 <div markdown="span" class="alert alert-primary">:bulb: **Extra Information**
 <br><br>
-- `GROUP`: A string representing the group of students as shown in the list. Must not contain spaces.
+- `GROUP`: A string representing the group of students as shown in the list. Must only be alphanumeric and cannot 
+contain space.
```
Comment (et-irl) [inline docs/UserGuide.md:504]: spaces

File: docs/UserGuide.md [hunk 4]
```diff
@@ -542,7 +545,8 @@ Unmark the attendance of a group of students for that tutorial.
 
 <div markdown="span" class="alert alert-primary">:bulb: **Extra Information**
 <br><br>
-- `GROUP`: A string representing the group of students as shown in the list. Must not contain space.
+- `GROUP`: A string representing the group of students as shown in the list. Must only be alphanumeric and cannot 
+contain space.
 <br>
```
Comment (et-irl) [inline docs/UserGuide.md:550]: spaces

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -566,7 +567,7 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 * 3a. The given index is invalid.
 
-    * 3a1. AddressBook shows an error message.
+    * 3a1. `npc_track`  shows an error message.
```
Comment (et-irl) [inline docs/DeveloperGuide.md:570]: extraneous space

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -552,9 +553,9 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 **MSS**
 
 1.  User requests to list persons
-2.  AddressBook shows a list of persons
+2.  `npc_track`  shows a list of persons
 3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+4.  `npc_track`  deletes the person
```
Comment (et-irl) [inline docs/DeveloperGuide.md:558]: extraneous space

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -552,9 +553,9 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 **MSS**
 
 1.  User requests to list persons
-2.  AddressBook shows a list of persons
+2.  `npc_track`  shows a list of persons
```
Comment (et-irl) [inline docs/DeveloperGuide.md:556]: extraneous space

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -136,7 +135,9 @@ The `Model` component,
 * stores a `UserPref` object that represents the user’s preferences. This is exposed to the outside as a `ReadOnlyUserPref` objects.
 * does not depend on any of the other three components (as the `Model` represents data entities of the domain, they should make sense on their own without depending on other components)
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** An alternative (arguably, a more OOP) model is given below. It has a `Tag` list in the `AddressBook`, which `Person` references. This allows `AddressBook` to only require one `Tag` object per unique tag, instead of each `Person` needing their own `Tag` objects.<br>
+<div markdown="span" class="alert alert-info">:information_source: **Note:** An alternative (arguably, a more OOP) 
+model is given below. It has a `Tag` list in the `NpcTrack`, which `Person` references. This allows `NpcTrack` to 
+only require one `Tag` object per unique tag, instead of each `Person` needing their own `Tag` objects.<br>
```
Comment (et-irl) [inline docs/DeveloperGuide.md:140]: npc_track over NpcTrack for consistency?

**Review by et-irl — COMMENTED** [review]
grammar

**Pull request merged**